dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -959,57 +950,24 @@ def wait_for_jquery
   }
 end
 
-def enroll_in_plc_course(user_email)
-  require_rails_env
-  user = User.find_by_email_or_hashed_email(user_email)
-  course = Course.find_by(name: 'All The PLC Things')
-  enrollment = Plc::UserCourseEnrollment.create(user: user, plc_course: course.plc_course)
-  enrollment.plc_unit_assignments.update_all(status: Plc::EnrollmentUnitAssignment::IN_PROGRESS)
-end
-
 Given(/^I am enrolled in a plc course$/) do
-  enroll_in_plc_course(@users.first[1][:email])
-end
-
-def create_user(**args)
-  name = "Fake User"
-  email, password = generate_user(name)
-  attributes = {
-    name: name,
-    email: email,
-    password: password,
-    user_type: "teacher",
-    age: "21+"
-  }.merge!(args)
-  user = User.new(attributes)
-  user.save ? user : nil
-end
-
-def assign_script_as_student(user_email, script_name)
-  require_rails_env
-  script = Script.find_by_name(script_name)
-  section = Section.create(name: "New Section", user: create_user, script: script)
-  user = User.find_by_email_or_hashed_email(user_email)
-  section.students << user
+  browser_request(url: '/api/test/enroll_in_plc_course', method: 'POST')
 end
 
 Given(/^I am assigned to script "([^"]*)"$/) do |script_name|
-  assign_script_as_student(@users.first[1][:email], script_name)
+  browser_request(
+    url: '/api/test/assign_script_as_student',
+    method: 'POST',
+    body: {script_name: script_name}
+  )
 end
 
 Then(/^I fake completion of the assessment$/) do
-  user = User.find_by_email_or_hashed_email(@users.first[1][:email])
-  unit_assignment = Plc::EnrollmentUnitAssignment.find_by(user: user)
-  unit_assignment.enroll_user_in_unit_with_learning_modules(
-    [
-      unit_assignment.plc_course_unit.plc_learning_modules.find_by(module_type: Plc::LearningModule::CONTENT_MODULE),
-      unit_assignment.plc_course_unit.plc_learning_modules.find_by(module_type: Plc::LearningModule::PRACTICE_MODULE)
-    ]
-  )
+  browser_request(url: '/api/test/fake_completion_assessment', method: 'POST', code: 204)
 end
 
 def generate_user(name)
-  email = "user#{Time.now.to_i}_#{rand(1000)}@testing.xx"
+  email = "user#{Time.now.to_i}_#{rand(1_000_000)}@test.xx"
```

Comment:
Generate a little more randomness here, to prevent the likelihood of 'email already in use' collisions when running multiple tests concurrently.